src/test/java/baron/Utils/DateUtilsTest.java

```diff
@@ -0,0 +1,62 @@
+package baron.Utils;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class DateUtilsTest {
+    @Test
+    public void parseDateString_correctDateFormat_dateReturned() {
+        String dateString = "9/10/1999 1600";
+        LocalDateTime actual = DateUtils.parseDate(dateString);
+        LocalDateTime expected = LocalDateTime.of(1999, 10, 9, 16, 0, 0);
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void parseDateString_incorrectDateFormat_dateReturned() {
+        // Invalid day
+        assertThrows(IllegalArgumentException.class, () -> {
+            String dateString = "32/10/1999 1600";
+            LocalDateTime actual = DateUtils.parseDate(dateString);
+        });
+
+        // Invalid month
+        assertThrows(IllegalArgumentException.class, () -> {
+            String dateString = "1/13/1999 1600";
+            LocalDateTime actual = DateUtils.parseDate(dateString);
+        });
+
+        // Invalid year
+        assertThrows(IllegalArgumentException.class, () -> {
+            String dateString = "1/13/199 1600";
+            LocalDateTime actual = DateUtils.parseDate(dateString);
+        });
+    }
```

Comment:
I like how your test cases are clear and elaborate and most importantly cover all edge cases that might happen, the detailing is elaborate and concise! 👍